service/service.go

```diff
@@ -124,6 +126,14 @@ func FileLoaderConfigFactory(v *viper.Viper, factories component.Factories) (*co
 	if err != nil {
 		return nil, fmt.Errorf("error loading config file %q: %v", file, err)
 	}
+	for _, k := range v.AllKeys() {
+		v.Set(k, v.Get(k))
+	}
```

Comment:
Why do we do this? Code is not clear and looks suspect (we get and set all keys?). Add comment.

Reply:
I have added a comment